### PR TITLE
Fix env, fix variables and safety checks for pm2/nodemon/docker modes

### PR DESCRIPTION
**Pull request by davidzwa**
isDocker safe fixes:
- [x] Env file now created if not exists
- [x] Port written if not set
- [x] MongoDB url written if not set

env utils:
- [x] will not write variables when in docker mode (not tested)

constants:
- [x] octofarm port env key
- [x] mongo env key
- [x] non npm mode env key
- [x] prerelease boolean key

fixed tests + bug:
- [x] database default url now correctly provided to template
- [x] fix timeout issue (startup slow with some runners)

Tested by clearing the .env keys and restart

### CODE REVIEW

**Comment by NotExpectedYet** [comment]
Failing v14

File: app-env.js [hunk 1]
```diff
@@ -69,47 +76,116 @@ function removeDeprecatedMongoURIConfigFile() {
...
+function fetchMongoDBConnectionString(persistToEnv = false) {
+  if (!process?.env[AppConstants.MONGO_KEY]) {
+    logger.warning(
+      `~ ${AppConstants.MONGO_KEY} environment variable is not set. Assuming default: ${AppConstants.MONGO_KEY}=${AppConstants.defaultMongoStringUnauthenticated}`
+    );
+    printInstructionsURL();
+    process.env[AppConstants.MONGO_KEY] = AppConstants.defaultMongoStringUnauthenticated;
+
+    // is not isDocker just to be sure
+    if (persistToEnv && !isDocker()) {
```
[the remark this answers is not in the document]
Comment (davidzwa) [reply app-env.js:88]: Did the latter.

File: app-env.js [hunk 2]
```diff
@@ -69,47 +76,116 @@ function removeDeprecatedMongoURIConfigFile() {
...
+function fetchOctoFarmPort() {
+  const port = process.env[AppConstants.OCTOFARM_PORT_KEY];
+  if (Number.isNaN(parseInt(port))) {
+    logger.warning(
+      `~ The ${AppConstants.OCTOFARM_PORT_KEY} setting was not a correct port number: >= 0 and < 65536. Actual value: ${port}.`
+    );
+
+    // is not isDocker just to be sure
+    if (!isDocker()) {
+      envUtils.writeVariableToEnvFile(
```
[the remark this answers is not in the document]
Comment (davidzwa) [reply app-env.js:108]: Did the latter.

File: server_src/lib/logger.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ class LoggerService {
   }
 
   warning(message, obj) {
-    this.logger.log("warning", message, {
+    this.logger.log("warn", message, {
```
[the remark this answers is not in the document]
Comment (NotExpectedYet) [reply server_src/lib/logger.js:61]: :p

File: server_src/routes/databaseIssue.js [hunk 4]
```diff
@@ -120,7 +123,7 @@ router.post("/save-connection-env", async (req, res) => {
   }
 
   try {
-    envUtils.writeVariableToEnvFile(path.join(__dirname + "/../../.env"), "MONGO", connectionURL);
+    envUtils.writeVariableToEnvFile(path.join(__dirname, "../../.env"), AppConstants.MONGO_KEY, connectionURL);
```
[the remark this answers is not in the document]
Comment (NotExpectedYet) [reply server_src/routes/databaseIssue.js:126]: I'm not sure tbh, couldn't you pass through the rootDir?

**Review by NotExpectedYet — APPROVED** [review]
All good to me, couldn't break and like the new re-arranging. 

I'll merge, the dirname I gather we can sort later.

**Pull request merged**